### PR TITLE
Ni/simpl init

### PR DESCRIPTION
**Pull request by nehabluedot**
The test can be done by adding Prod ProjectID and valid destinationID for a project

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@
 /app/build
 /.idea
 /.gradle
-*.iml
\ No newline at end of file
+*.iml
+gradle.properties
```
Comment (jsprincep) [inline .gitignore]: That's a bit extreme, now anyone who checks this out will have to add the gradle.properties file with the useAndroidX and whatnot in it. Better to simply not commit the code for accessing our private maven repo in `build.gradle`.

File: app/src/main/java/au/com/bluedot/minimalintegration/AppGeoTriggerReceiver.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import android.widget.Toast;
+import au.com.bluedot.point.net.engine.GeoTriggeringEventReceiver;
+import au.com.bluedot.point.net.engine.ZoneEntryEvent;
+import au.com.bluedot.point.net.engine.ZoneExitEvent;
+import au.com.bluedot.point.net.engine.ZoneInfo;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+public class AppGeoTriggerReceiver extends GeoTriggeringEventReceiver {
+    private final Handler handler = new Handler(Looper.getMainLooper());
```
Comment (jsprincep) [inline app/src/main/java/au/com/bluedot/minimalintegration/AppGeoTriggerReceiver.java]: Shouldn't be any need for this, broadcasts are always received on the main thread.

File: app/src/main/java/au/com/bluedot/minimalintegration/AppTempoReceiver.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import android.os.Looper;
+import android.widget.Toast;
+import au.com.bluedot.point.net.engine.BDError;
+import au.com.bluedot.point.net.engine.TempoTrackingReceiver;
+import org.jetbrains.annotations.NotNull;
+
+public class AppTempoReceiver extends TempoTrackingReceiver {
+    private final Handler handler = new Handler(Looper.getMainLooper());
+
+    /**
```
Comment (jsprincep) [inline app/src/main/java/au/com/bluedot/minimalintegration/AppTempoReceiver.java:12]: I'm not against adding the documentation in this project, but we should be consistent - why here and not in the geotriggerReceiver?

File: app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java [hunk 4]
```diff
@@ -62,17 +44,27 @@ public void initPointSDK() {
...
+                        text = text + "Success ";
+                        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+                        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                        startActivity(intent);
+                    }
+                    Toast.makeText(getApplicationContext(),text,Toast.LENGTH_LONG).show();
+                };
+                //ProjectID for the App

+                String projectId = "<YOUR-PROJECT-ID>";
```
Comment (jsprincep) [inline app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java]: Let's make this a constant right at the top of the file so it's harder to miss

File: app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java [hunk 5]
```diff
@@ -85,138 +77,53 @@ private void requestPermissions() {
...
-            @Override
-            public void run() {
-                Toast.makeText(getApplicationContext(), "Left: " + beaconInfo.getName() + " dwellTime,min=" + dwellTime, Toast.LENGTH_LONG)
-                        .show();
-            }
-        });
+    void startTempo(){
+        TempoService.builder()
+                .notification(createNotification())
+                .destinationId("<DESTINATION_ID>")
```
Comment (jsprincep) [inline app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java]: Let's also make the tempo destinationId a constant at the top of the file with a comment.

File: app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java [hunk 6]
```diff
@@ -249,16 +155,24 @@ private Notification createNotification() {
...
-                    .setCategory(Notification.CATEGORY_SERVICE)
                     .setPriority(PRIORITY_MAX)
                     .setSmallIcon(R.mipmap.ic_launcher);
-
             return notification.build();
         }
     }
+
+    @Override public void onTempoResult(@Nullable BDError bdError) {
+        String text = "Tempo start";
```
Comment (jsprincep) [inline app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java]: Not sure the string concatenation is really saving us much here, but we should at least put a space between start and the next word.

File: build.gradle [hunk 7]
```diff
@@ -4,13 +4,23 @@ buildscript {
...
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.6.3'
+        classpath 'com.android.tools.build:gradle:4.0.0'
     }
 }
 
 allprojects {
     repositories {
         google()
+        maven {
```
Comment (jsprincep) [inline build.gradle]: As mentioned above, better to not commit this and to include the gradle.properties without the token.

**Review by jsprincep — CHANGES_REQUESTED** [review]
Looking good, just a few things to fix up.

File: .gitignore (hunk 1 above)
Comment (nehabluedot) [reply .gitignore]: fixed it

File: build.gradle (hunk 7 above)
Comment (nehabluedot) [reply build.gradle]: updated the PR with changes

File: app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java (hunk 4 above)
Comment (nehabluedot) [reply app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java]: moved projectId and destinationId to the top of the class

File: app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java (hunk 6 above)
Comment (nehabluedot) [reply app/src/main/java/au/com/bluedot/minimalintegration/MainApplication.java]: fixed it

File: app/build.gradle [hunk 8]
```diff
@@ -26,5 +26,6 @@ android {
 
 dependencies {
     implementation 'androidx.appcompat:appcompat:1.1.0'
-    implementation 'com.gitlab.bluedotio.android:point_sdk_android:15.2.0'
+    implementation "au.com.bluedot:pointsdk:229200004"
+   // implementation 'com.gitlab.bluedotio.android:point_sdk_android:15.2.0'
```
Comment (jsprincep) [inline app/build.gradle]: Can use 15.3.0 release now :)

**Review by jsprincep — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 8 above)
Comment (nehabluedot) [reply app/build.gradle]: Updated to 15.3.0

**Review by jsprincep — APPROVED** [review]
🥳

**Pull request merged**